api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -350,4 +355,127 @@ private OrderSet orderSetBuilder(boolean orderSetRetired, boolean orderSetMember
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		assertNull(Context.getOrderSetService().getOrderSetAttributeTypeByUuid("not-a-uuid"));
+	}
+
+	/**
+	 * @see OrderSetService#purgeOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void purgeOrderSetAttributeType_shouldCompletelyRemoveAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(2, initialOrderSetAttributeTypesCount);
+		Context.getOrderSetService().purgeOrderSetAttributeType(Context.getOrderSetService().getOrderSetAttributeType(2));
+		assertEquals(initialOrderSetAttributeTypesCount - 1, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#retireOrderSetAttributeType(OrderSetAttributeType,String)
+	 */
+	@Test
+	public void retireOrderSetAttributeType_shouldRetireAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetAttributeType orderSetAttributeType = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertFalse(orderSetAttributeType.getRetired());
+		assertNull(orderSetAttributeType.getRetiredBy());
+		assertNull(orderSetAttributeType.getDateRetired());
+		assertNull(orderSetAttributeType.getRetireReason());
+		Context.getOrderSetService().retireOrderSetAttributeType(orderSetAttributeType, "for testing");
+		orderSetAttributeType = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertTrue(orderSetAttributeType.getRetired());
+		assertNotNull(orderSetAttributeType.getRetiredBy());
+		assertNotNull(orderSetAttributeType.getDateRetired());
+		assertEquals("for testing", orderSetAttributeType.getRetireReason());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldCreateANewOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(2, initialOrderSetAttributeTypesCount);
+		OrderSetAttributeType orderSetAttributeType = new OrderSetAttributeType();
+		orderSetAttributeType.setName("Another one");
+		orderSetAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		Context.getOrderSetService().saveOrderSetAttributeType(orderSetAttributeType);
+		assertNotNull(orderSetAttributeType.getId());
+		assertEquals(initialOrderSetAttributeTypesCount + 1, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldEditAnExistingOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetService service = Context.getOrderSetService();
+		assertEquals(2, service.getAllOrderSetAttributeTypes().size());
```

Comment:
Problem with hard coding the number as 2 is that when the test database gets changed, the test will fail.

Reply:
thanks ,,,going to to fix it